chef.py

```diff
@@ -0,0 +1,156 @@
+
+BASE_URL = 'http://www.prathamopenschool.org'
+# In debug mode, only one channel is downloaded.
+DEBUG_MODE = False
+
+
+def construct_channel(*args, **kwargs):
+    channel = ChannelNode(
+        title='Pratham Open School',
+        source_domain=BASE_URL,
+        source_id='pratham-open-school',
+    )
+    global DEBUG_MODE
+    DEBUG_MODE = 'debug' in kwargs
+    get_topics(channel, kwargs['language'])
+    return channel
+
+
+def get_topics(parent, path):
+    doc = get_page(path)
+    try:
+        menu_row = doc.find('div', {'id': 'menu-row'})
+    except Exception as e:
+        LOGGER.error('get_topics: %s : %s' % (e, doc))
+    for topic in menu_row.find_all('a'):
+        try:
+            if topic['href'] == '#':
+                continue
+            LOGGER.info('topic: %s' % (topic['href']))
+            title = topic.get_text().strip()
+            source_id = get_source_id(topic['href'])
+            node = TopicNode(title=title, source_id=source_id)
+            parent.add_child(node)
+            get_subtopics(node, topic['href'])
+            if DEBUG_MODE:
+                return
+        except Exception as e:
+            LOGGER.error('get_topics: %s : %s' % (e, topic))
+
+
+def get_subtopics(parent, path):
+    doc = get_page(path)
+    try:
+        menu_row = doc.find('div', {'id': 'body-row'})
+        menu_row = menu_row.find('div', {'class': 'col-md-2'})
+    except Exception as e:
+        LOGGER.error('get_subtopics: %s : %s' % (e, doc))
+    for subtopic in menu_row.find_all('a'):
+        try:
+            LOGGER.info('subtopic: %s' % (subtopic['href']))
+            title = subtopic.get_text().strip()
+            source_id = get_source_id(subtopic['href'])
+            node = TopicNode(title=title, source_id=source_id)
+            parent.add_child(node)
+            get_lessons(node, subtopic['href'])
+        except Exception as e:
+            LOGGER.error('get_subtopics: %s : %s' % (e, subtopic))
+
+
+def get_lessons(parent, path):
+    doc = get_page(path)
+    try:
+        menu_row = doc.find('div', {'id': 'body-row'})
+        menu_row = menu_row.find('div', {'class': 'col-md-9'})
+    except Exception as e:
+        LOGGER.error('get_lessons: %s : %s' % (e, doc))
+    for lesson in menu_row.find_all('div', {'class': 'thumbnail'}):
+        try:
+            title = lesson.find('div', {'class': 'txtline'}).get_text().strip()
+            link = lesson.find('a')['href']
+            LOGGER.info('lesson: %s' % (link))
+            source_id = get_source_id(link)
+            node = TopicNode(title=title, source_id=source_id)
+            parent.add_child(node)
+            get_contents(node, link)
+        except Exception as e:
+            LOGGER.error('get_lessons: %s : %s' % (e, lesson))
+
+
+def get_contents(parent, path):
+    doc = get_page(path)
+    try:
+        menu_row = doc.find('div', {'id': 'row-exu'})
+    except Exception as e:
+        LOGGER.error('get_contents: %s : %s' % (e, doc))
```

Comment:
May want to return after this, if you want it to "gracefully fail", since line 117 won't work if line 114 errored out?